main/src/library/Int32.flix

```diff
@@ -279,13 +279,12 @@ namespace Int32 {
     /// Returns the factorial of `x`.
     ///
     @Time(factorial(x)) @Space(1)
-    pub def factorial(x: Int32): Int32 = factorialHelper(x, 1)
-
-    ///
-    /// Helper function for `factorial`.
-    ///
-    def factorialHelper(x: Int32, acc: Int32): Int32 =
-        if (x == 0) acc else factorialHelper(x - 1, x * acc)
+    pub def factorial(x: Int32): Int32 =
+        def factorialHelper(y, acc) = match y {
```

Comment:
Looks good!

By convention these helper functions are now always called `loop` throughout the standard library.

Reply:
resolved by 60cd15b22c0b88b46b579a56a156336fe33dbc7b :+1: